packages/bot-web-ui/src/components/dashboard/bot-builder/__tests__/bot-builder.spec.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { render, screen } from '@testing-library/react';
+import BotBuilder from '..';
+import { useDBotStore } from 'Stores/useDBotStore';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => ({
+    saveRecentWorkspace: jest.fn(),
+    unHighlightAllBlocks: jest.fn(),
+    terminateBot: jest.fn(),
+    terminateConnection: jest.fn(),
+    initWorkspace: jest.fn().mockResolvedValue({ key: 0 }),
+}));
+
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+jest.mock('../../dbot-tours/bot-builder-tour', () => jest.fn(() => <div>BotBuilderTourHandler</div>));
+jest.mock('../../../quick-strategy', () => jest.fn(() => <div>QuickStrategy1</div>));
+jest.mock('../workspace-wrapper', () => jest.fn(() => <div>WorkspaceWrapper</div>));
+jest.mock('../../dashboard-component/load-bot-preview/save-modal', () => jest.fn(() => <div>SaveModal</div>));
+jest.mock('../../../load-modal', () => jest.fn(() => <div>LoadModal</div>));
+jest.mock('../../../bot-snackbar', () =>
+    jest.fn(({ handleClose }) => <div onClick={() => handleClose()}>BotSnackbar</div>)
+);
+
+const mockDBot = {
+    dashboard: {
+        is_preview_on_popup: false,
+        active_tab: 0,
+        active_tour: false,
+    },
+    quick_strategy: {
+        is_open: true,
+    },
+    run_panel: {
+        is_running: true,
+    },
+    app: {
+        onMount: jest.fn(),
+        onUnmount: jest.fn(),
+    },
+};
+
+jest.mock('../../../../stores/useDBotStore', () => ({
+    useDBotStore: jest.fn(() => mockDBot),
+}));
```

Comment:
P.S. I had to add this mock in order to handle more cases, because `is_preview_on_popup` has a reaction and when the value equals true, automatically becomes false.
<img width="380" alt="Screenshot 2023-12-04 at 20 58 14" src="https://github.com/binary-com/deriv-app/assets/103181650/e120c4d9-06f6-49ac-856f-399124e238a5">